modules/core/src/lib/deck.js

```diff
@@ -436,32 +441,55 @@ export default class Deck {
     return views;
   }
 
-  _pickAndCallback(options) {
-    const pos = options.event.offsetCenter;
-    // Do not trigger callbacks when click/hover position is invalid. Doing so will cause a
-    // assertion error when attempting to unproject the position.
-    if (!pos) {
-      return;
+  _requestPick({event, callback, mode, immediate}) {
+    const {_pickRequest} = this;
+    if (event.type === 'pointerleave') {
+      _pickRequest.x = -1;
+      _pickRequest.y = -1;
+      _pickRequest.radius = 0;
+    } else {
+      const pos = event.offsetCenter;
+      // Do not trigger callbacks when click/hover position is invalid. Doing so will cause a
+      // assertion error when attempting to unproject the position.
+      if (!pos) {
+        return;
+      }
+      _pickRequest.x = pos.x;
+      _pickRequest.y = pos.y;
+      _pickRequest.radius = this.props.pickingRadius;
     }
 
-    const radius = this.props.pickingRadius;
-    const layers = this.layerManager.getLayers();
-    const activateViewport = this.layerManager.activateViewport;
-    const selectedInfos = this.deckPicker.pickObject({
-      x: pos.x,
-      y: pos.y,
-      radius,
-      layers,
-      viewports: this.getViewports(pos),
-      activateViewport,
-      mode: options.mode,
-      depth: 1,
-      event: options.event
-    });
-    if (options.callback && selectedInfos) {
-      const firstInfo = selectedInfos.find(info => info.index >= 0) || null;
-      // As per documentation, send null value when no valid object is picked.
-      options.callback(firstInfo, selectedInfos, options.event.srcEvent);
+    _pickRequest.callback = callback;
+    _pickRequest.event = event;
+    _pickRequest.mode = mode;
+
+    if (immediate) {
+      this._pickAndCallback();
+    }
+  }
+
+  _pickAndCallback() {
+    const {_pickRequest} = this;
+
+    if (_pickRequest.mode) {
```

Comment:
Is there any reason to pick if a callback isn't provided?

Reply:
Yes, because there are callback props in each layer as well. `autoHighlight` also depends on picking.